src/chain/chain.rs

```diff
@@ -695,24 +711,38 @@ impl Chain {
 
     /// Returns whether the given peer is elder in our section.
     pub fn is_peer_our_elder(&self, pub_id: &PublicId) -> bool {
-        self.state.our_info().members().contains(pub_id)
-            || self.state.new_info.members().contains(pub_id)
+        self.state.our_info().is_member(pub_id) || self.state.new_info.is_member(pub_id)
     }
 
     /// Returns whether the given peer is elder in one of our neighbour sections.
     pub fn is_peer_neighbour_elder(&self, pub_id: &PublicId) -> bool {
-        self.neighbour_infos()
-            .any(|info| info.members().contains(pub_id))
+        self.neighbour_infos().any(|info| info.is_member(pub_id))
     }
 
     /// Returns elders from our own section according to the latest accumulated `SectionInfo`.
     pub fn our_elders(&self) -> impl Iterator<Item = &P2pNode> + ExactSizeIterator {
-        self.state.our_info().p2p_members().iter()
+        self.state.our_info().member_nodes()
     }
 
     /// Returns all neighbour elders.
-    pub fn neighbour_elders_p2p(&self) -> impl Iterator<Item = &P2pNode> {
-        self.neighbour_infos().flat_map(EldersInfo::p2p_members)
+    pub fn neighbour_elder_nodes(&self) -> impl Iterator<Item = &P2pNode> {
+        self.neighbour_infos().flat_map(EldersInfo::member_nodes)
+    }
+
+    /// Returns the elders for a neighbour section.
+    /// Returns None if the `Prefix` provided wasn't our own section or a neigbour.
+    pub fn get_section_elders(
+        &self,
+        names: &Prefix<XorName>,
+    ) -> Option<&BTreeMap<XorName, P2pNode>> {
+        if self.our_prefix() == names {
+            Some(self.our_info().member_map())
```

Comment:
Is this correct? Should it be from `our_info` or new_info`?

Reply:
our_info seem to make more sense.
new_info was essentially a way to handle remove and add concurrently, it should probably be removed, hopefully sometime soon.